src/olympia/lib/crypto/packaged.py

```diff
@@ -55,42 +57,81 @@ def get_id(addon):
     return hashlib.sha256(guid).hexdigest()
 
 
-def call_signing(file_obj, endpoint):
+def call_signing(file_obj):
     """Get the jar signature and send it to the signing server to be signed."""
     # Extract jar signature.
     jar = JarExtractor(path=storage.open(file_obj.file_path))
 
     log.debug(u'File signature contents: {0}'.format(jar.signatures))
 
-    log.debug(u'Calling signing service: {0}'.format(endpoint))
-    with statsd.timer('services.sign.addon'):
-        response = requests.post(
-            endpoint,
-            timeout=settings.SIGNING_SERVER_TIMEOUT,
-            data={'addon_id': get_id(file_obj.version.addon)},
-            files={'file': (u'mozilla.sf', unicode(jar.signatures))})
-    if response.status_code != 200:
-        msg = u'Posting to add-on signing failed: {0}'.format(response.reason)
+    use_autograph = waffle.switch_is_active('activate-autograph-signing')
+    signed_manifest = unicode(jar.signatures)
+    has_error = False
+
+    if use_autograph:
+        conf = settings.AUTOGRAPH_CONFIG
+        log.debug('Calling autograph service: {0}'.format(conf['server_url']))
+
+        # create the signing request
+        signing_request = [{
+            'input': b64encode(signed_manifest),
+            'keyid': conf['signer'],
+            'options': {
+                'id': get_id(file_obj.version.addon),
+            },
+        }]
+
+        # post the request
+        with statsd.timer('services.sign.addon'):
+            response = requests.post(
+                '{server}/sign/data'.format(server=conf['server_url']),
+                json=signing_request,
+                auth=HawkAuth(id=conf['user_id'], key=conf['key']))
+
+        if response.status_code != requests.codes.CREATED:
+            has_error = True
+    else:
+        log.debug(u'Calling signing service: {0}'.format(
+            settings.SIGNING_SERVER))
+
+        with statsd.timer('services.sign.addon'):
+            response = requests.post(
+                get_trunion_endpoint(settings.SIGNING_SERVER),
+                timeout=settings.SIGNING_SERVER_TIMEOUT,
+                data={'addon_id': get_id(file_obj.version.addon)},
+                files={'file': (u'mozilla.sf', signed_manifest)})
+
+        if response.status_code != requests.codes.OK:
+            has_error = True
+
+    if has_error:
+        msg = u'Posting to add-on signing failed: {0} {1}'.format(
+            response.reason, response.text)
         log.error(msg)
         raise SigningError(msg)
 
-    pkcs7 = b64decode(json.loads(response.content)['mozilla.rsa'])
+    # convert the base64 encoded pkcs7 signature back to binary
+    if use_autograph:
+        pkcs7 = urlsafe_b64decode(force_bytes(response.json()[0]['signature']))
```

Comment:
The signature is encoded in base64, not urlsafe_base64. https://github.com/mozilla-services/autograph/tree/master/signer/xpi#signature-response

Reply:
Damn, I forgot to change that from when I was accidentally using the content signing. Thanks!